xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -4914,6 +4914,9 @@ void CVideoPlayer::OnResetDisplay()
   m_VideoPlayerVideo->SendMessage(new CDVDMsgBool(CDVDMsg::GENERAL_PAUSE, false), 1);
   m_clock.Pause(false);
   m_displayLost = false;
+
+  if (m_VideoPlayerAudio)
+    dynamic_cast<CVideoPlayerAudio*>(m_VideoPlayerAudio)->TriggerCheckPassthrough();
```

Comment:
Not sure how much I like this cast.
Extending the interface `IDVDStreamPlayerAudio` for this feels weird/wrong as well.
Probably a new `CDVDMsg` would be justified?

Reply:
Updated to use the new` CDVDMsg`. In case we choose this option I will squash in a single commit.

Reply:
I like it better this way, but maybe @FernetMenta and/or @fritsch can give us their opinion?

Reply:
Mainly the infrastructure is already in place here. The above message / trigger is received cause video player subscribes to the windowing infrastructure. For the dynamic_cast, it could return NULL. If the direct method call is used as interface to establish business logic - add it to the interface, then the cast can be avoided.

The below message comit might be a better idea, but is done asychroneously.

As you see: I don't have a clear opinion. I see the need and understand the code that wants to accomplish that.

Reply:
> The below message comit might be a better idea, but is done asychroneously.

I think that in this case, asynchronous, it is also an advantage. If this adds a small delay it is better since it will be re-evaluate passthrough capabilities when the display and audio device are well seated (it could be the case that the display is already available and the audio not yet). With a little delay this possibility is reduced.

Reply:
> I like it better this way, but maybe @FernetMenta and/or @fritsch can give us their opinion?

Involving VideoPlayer in the chain of events isn't the clearest solution and does not cover all cases. A change of refresh rate triggers an audio device change event but device change events can occur without a change of refresh rate. The latter case does still not work.
Further re-evaluation of passthrough needs to be done when ActiveAE has completed its reconfiguration after a device change event. Otherwise codec factory might still query outdated capabilities.
PAPlayer needs a solution too.

A proper solution needs to make sure that AE has re-enumerated sinks and adjusted settings: https://github.com/xbmc/xbmc/blob/1fff2cfa6740dae2f0bb501d15558b92fc3befd2/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp#L629

Reply:
Apart from the fact that not all cases are covered this solution is not bad. I agree with @thexai that asynchronous messaging here is much better than the first approach.

Reply:
I agree that it does not cover "all" cases but it does cover the case that is causing problems to users: "Audio passthrough does not start initially if Adjust Display Refresh Rate is enabled".

It is preferable to have a "good" solution now than a "perfect" solution that does not exist yet. 😃

Reply:
In any case I think the functionality could be extended later if needed:

It is enough to send the message to `m_VideoPlayerAudio` in different ways.

Reply:
Async without clear sync points makes code non deterministic, that's why I stated it. Obviously a message properly sent helps a lot with e.g. concurrency.